test/pkg/environment/azure/identity.go

```diff
@@ -65,3 +69,106 @@ func (env *Environment) GetCurrentUserPrincipalID(ctx context.Context, cred azco
+	return &resp.Identity
+}
+
+// ExpectUpdatedManagedClusterKubeletIdentityAsync updates the kubelet identity of a managed cluster asynchronously
+// Returns the poller so the caller can control when to wait for completion
+func (env *Environment) ExpectUpdatedManagedClusterKubeletIdentityAsync(ctx context.Context, newIdentity *armmsi.Identity) *runtime.Poller[containerservice.ManagedClustersClientCreateOrUpdateResponse] {
+	GinkgoHelper()
+
+	By("getting current managed cluster configuration")
+	mc := env.ExpectGetManagedCluster()
+
+	By("updating kubelet identity")
+
+	// Update the kubelet identity in the identity profile
+	if mc.Properties.IdentityProfile == nil {
+		mc.Properties.IdentityProfile = make(map[string]*containerservice.UserAssignedIdentity)
+	}
+
+	mc.Properties.IdentityProfile["kubeletidentity"] = &containerservice.UserAssignedIdentity{
+		ClientID:   newIdentity.Properties.ClientID,
+		ObjectID:   newIdentity.Properties.PrincipalID,
+		ResourceID: newIdentity.ID,
+	}
+
+	// Update the cluster and return the poller
+	poller, err := env.managedClusterClient.BeginCreateOrUpdate(ctx, env.ClusterResourceGroup, env.ClusterName, mc, nil)
+	Expect(err).ToNot(HaveOccurred())
+
+	return poller
+}
+
+// ExpectGrantedACRAccess grants the specified identity access to pull from the ACR
+func (env *Environment) ExpectGrantedACRAccess(ctx context.Context, identity *armmsi.Identity) {
+	GinkgoHelper()
+
+	By("granting ACR pull access to identity")
+
+	// Get the ACR resource ID
+	acrResourceID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.ContainerRegistry/registries/%s",
+		env.SubscriptionID, env.ClusterResourceGroup, env.ACRName)
+
+	// AcrPull role definition ID: 7f951dda-4ed3-4680-a7ca-43fe172d538d
+	acrPullRoleID := fmt.Sprintf("/subscriptions/%s/providers/Microsoft.Authorization/roleDefinitions/7f951dda-4ed3-4680-a7ca-43fe172d538d", env.SubscriptionID)
+
+	identityPrincipalID := lo.FromPtr(identity.Properties.PrincipalID)
+
+	err := env.RBACManager.EnsureRoleWithPrincipalType(ctx, acrResourceID, acrPullRoleID, identityPrincipalID, "ServicePrincipal")
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// CheckClusterIdentityType returns the type of managed identity used by the cluster
+func (env *Environment) CheckClusterIdentityType(ctx context.Context) string {
+	mc := env.ExpectGetManagedCluster()
+	if mc.Identity == nil {
+		return "none"
+	}
+	if mc.Identity.Type == nil {
+		return "unknown"
+	}
+	return string(*mc.Identity.Type)
+}
+
+// IsClusterUserAssignedIdentity checks if the cluster uses user-assigned managed identity
+func (env *Environment) IsClusterUserAssignedIdentity(ctx context.Context) bool {
+	identityType := env.CheckClusterIdentityType(ctx)
+	return identityType == "UserAssigned"
+}
+
+// GetKubeletIdentityResourceID returns the current kubelet identity resource ID
+func (env *Environment) GetKubeletIdentityResourceID(ctx context.Context) string {
```

Comment:
minor, style:

These two methods could just be one `GetKubeletIdentity` that returns the identity object and the caller could just go `.ClientID` or `.ResourceID`.

Slightly less code and also works better if you end up wanting another field of the identity later